code/go/internal/validator/semantic/version_integrity_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestChangelogLinks(t *testing.T) {
+
+	var tests = []struct {
+		name      string
+		links     []string
+		numErrors int
+	}{
```

Comment:
Hm... I think you can also prepare table tests with given URLs and expected outcomes (nil or error).

Then, I would add another single unit test to check if `ensureLinksAreValid` iterates over all links (one test with few broken URLs and few valid ones).

Reply:
The code has been refactored to accomodate other link types and error handling.